create_jenkins_job.py

```diff
@@ -114,73 +116,89 @@ def main(argv=None):
         if os_name == 'linux':
-            job_name = 'nightly_' + os_name + '_coverage'
-            job_data['cmake_build_type'] = 'Debug'
-            job_data['enable_c_coverage_default'] = 'true'
-            job_config = expand_template('ci_job.xml.em', job_data)
-            configure_job(jenkins, job_name, job_config, **jenkins_kwargs)
-            job_data['enable_c_coverage_default'] = 'false'
+            create_job(os_name, 'nightly_linux_coverage', 'ci_job.xml.em', {
+                'cmake_build_type': 'Debug',
+                'enable_c_coverage_default': 'true',
+                'time_trigger_spec': PERIODIC_JOB_SPEC,
+                'mailer_recipients': DEFAULT_MAIL_RECIPIENTS,
+            })
 
         # configure nightly triggered job
-        job_name = 'nightly_' + os_name + '_release'
-        if os_name == 'win':
-            job_name = job_name[0:-4]
-        job_data['cmake_build_type'] = 'Release'
-        job_config = expand_template('ci_job.xml.em', job_data)
-        configure_job(jenkins, job_name, job_config, **jenkins_kwargs)
+        short_name = 'nightly_' + os_name + '_release'
+        if os_name == 'windows':
+            short_name = 'nightly_win_rel'
+        create_job(os_name, short_name, 'ci_job.xml.em', {
+            'cmake_build_type': 'Release',
+            'time_trigger_spec': PERIODIC_JOB_SPEC,
+            'mailer_recipients': DEFAULT_MAIL_RECIPIENTS,
+        })
 
         # configure nightly triggered job with repeated testing
-        job_name = 'nightly_' + os_name + '_repeated'
-        if os_name == 'win':
-            job_name = job_name[0:-5]
-        job_data['time_trigger_spec'] = '30 7 * * *'
-        job_data['cmake_build_type'] = 'None'
-        job_data['ament_test_args_default'] = '--retest-until-fail 20 --ctest-args -LE linter --'
-        job_config = expand_template('ci_job.xml.em', job_data)
-        configure_job(jenkins, job_name, job_config, **jenkins_kwargs)
+        short_name = 'nightly_' + os_name + '_repeated'
+        if os_name == 'windows':
+            short_name = 'nightly_win_rep'
+        create_job(os_name, short_name, 'ci_job.xml.em', {
+            'cmake_build_type': 'None',
+            'time_trigger_spec': PERIODIC_JOB_SPEC,
+            'mailer_recipients': DEFAULT_MAIL_RECIPIENTS,
+            'ament_test_args_default': '--retest-until-fail 20 --ctest-args -LE linter --',
+        })
+
+        # configure turtlebot jobs on Linux only for now
+        if os_name in ['linux', 'linux-aarch64']:
+            create_job(os_name, 'ci_turtlebot-demo_%s' % (os_name), 'ci_job.xml.em', {
+                'cmake_build_type': 'None',
+                'turtlebot_demo': True,
+                'supplemental_repos_url': 'https://raw.githubusercontent.com/ros2/turtlebot2_demo/master/turtlebot2_demo.repos',
+            })
+            create_job(os_name, 'nightly_turtlebot-demo_%s_debug' % (os_name), 'ci_job.xml.em', {
+                'cmake_build_type': 'Debug',
```

Comment:
Note: Cartographer doesn't build in `Debug` AFAIK so I would expect this job to fail every night